packages/synapse-interface/components/StateManagedBridge/FromTokenSelect.tsx

```diff
@@ -0,0 +1,84 @@
+import _ from 'lodash'
+import { useDispatch } from 'react-redux'
+import Select from 'react-select'
+
+import { Token } from '@/utils/types'
+import { setFromToken } from '@/slices/bridge/reducer'
+import { coinSelectStyles } from './styles/coinSelectStyles'
+import { useId } from 'react'
+
+import { useAppSelector } from '@/store/hooks'
+
+const ImageAndCoin = ({ option }: { option: Token }) => {
+  const { fromChainId } = useAppSelector((state) => state.bridge)
+  const { balancesAndAllowances } = useAppSelector((state) => state.portfolio)
```

Comment:
Can replace `useAppSelector((state) => state.bridge)`  with `useBridgeState()` 

and

Can utilize helper hook to directly pull `balancesAndAllowances` from Portfolio store
`
export const usePortfolioBalances = (): NetworkTokenBalancesAndAllowances => {
  return useAppSelector((state) => state.portfolio.balancesAndAllowances)
}
`